python/tests/test_eigen.py

```diff
@@ -0,0 +1,59 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+# Copyright (c) 2020 Scipp contributors (https://github.com/scipp)
+# @file
+# @author Neil Vaytet
+
+import numpy as np
+import scipp as sc
+
+
+def test_create_variable_0D_vector_3_float64():
+    var = sc.Variable(value=np.array([1, 2, 3]),
+                      unit=sc.units.m,
+                      dtype=sc.dtype.vector_3_float64)
+    np.testing.assert_array_equal(var.value, [1, 2, 3])
+    assert var.dtype == sc.dtype.vector_3_float64
+    assert var.unit == sc.units.m
+
+
+def test_create_variable_1D_vector_3_float64():
+    var = sc.Variable(dims=['x'],
+                      values=[[1, 2, 3], [4, 5, 6]],
+                      unit=sc.units.m,
+                      dtype=sc.dtype.vector_3_float64)
+    assert len(var.values) == 2
+    np.testing.assert_array_equal(var.values[0], [1, 2, 3])
+    np.testing.assert_array_equal(var.values[1], [4, 5, 6])
+    assert var.dims == ['x']
+    assert var.dtype == sc.dtype.vector_3_float64
+    assert var.unit == sc.units.m
+
+
+def test_create_quaternion_float64():
+    quat = sc.Quat(np.arange(4.0))
+    assert quat.x() == 0.0
+    assert quat.y() == 1.0
+    assert quat.z() == 2.0
+    assert quat.w() == 3.0
```

Comment:
The order in the constructor of `Eigen::Quaternion` is w, x, y, z. Here it is different. I could not see where it goes wrong, but it must be, somewhere?

Reply:
After casting to a `vector<vector<double>>`, I can then use the constructor that takes in a pointer to a vector.